docs/pages/api-docs/data-grid/x-grid.md

```diff
@@ -117,40 +117,35 @@ The `ref` is forwarded to the root element.
 
 ## Slots
 
-Api of the `components` props of type `GridSlotsComponent`
+You can use the [slots API](/components/data-grid/components/#overriding-components) to override nested components or to inject extra props.
 
 | Name | Type | Default | Description |
 |:-----|:-----|:--------|:------------|
 | <span class="prop-name">Checkbox</span> | <span class="prop-type">elementType</span> | <span class="prop-type">Checkbox</span> | Checkbox component used in the grid for both header and cells. Default it uses the Material UI core Checkbox component.|
+| <span class="prop-name">ColumnFilteredIcon</span> | <span class="prop-type">elementType </span> | <span class="prop-type">FilterAltIcon</span> | Icon displayed on the column header menu to show that a filer has been applied to the column. |
 | <span class="prop-name">ColumnMenu</span> | <span class="prop-type">elementType&lt;GridColumnMenuProps></span> | <span class="prop-type">GridColumnMenu</span> | Column menu component rendered by clicking on the 3 dots "kebab" icon in column headers.|
+| <span class="prop-name">ColumnMenuIcon</span> | <span class="prop-type">elementType </span> | <span class="prop-type">TripleDotsVerticalIcon</span> | Icon displayed on the side of the column header title to display the filter input component. |
+| <span class="prop-name">ColumnResizeIcon</span> | <span class="prop-type">elementType </span> | <span class="prop-type">SeparatorIcon</span> |  Icon displayed in between two column headers that allows to resize the column header. |
+| <span class="prop-name">ColumnSelectorIcon</span> | <span class="prop-type">elementType </span> | <span class="prop-type">ColumnIcon</span> | Icon displayed on the column menu selector tab. |
+| <span class="prop-name">ColumnSortedAscendingIcon</span> | <span class="prop-type">elementType </span> | <span class="prop-type">ArrowUpwardIcon</span> | Icon displayed on the side of the column header title when sorted in Ascending order. |
+| <span class="prop-name">ColumnSortedDescendingIcon</span> | <span class="prop-type">elementType </span> | <span class="prop-type">ArrowDownwardIcon</span> | Icon displayed on the side of the column header title when sorted in Descending order.|
 | <span class="prop-name">ColumnsPanel</span> | <span class="prop-type">elementType</span> | <span class="prop-type">ColumnsPanel</span> | GridColumns panel component rendered when clicking the columns button.|
+| <span class="prop-name">ColumnUnsortedIcon</span> | <span class="prop-type">React.ElementType<br>&#124;&nbsp;null </span> | <span class="prop-type">GridColumnUnsortedIcon</span> | Icon displayed on the side of the column header title when unsorted. |
```

Comment:
This one is inconsistent with the others `React.ElementType | null` vs `elementType`

Reply:
The slot is optional, providing null changes the behavior. But yeah, I think that we can use the opportunity to normalize further https://github.com/mui-org/material-ui/issues/27274 at the same time

Reply:
But yeah, we can do a follow-up on my changes, I didn't want to push it too far. We could, for instance:

- have a single object structure, instead of 3 (GridIconSlotsComponent, GridApiRefComponentsProperty, GridSlotsComponent)
- uniformized the `components` prop
- uniformized the `componentsProps` prop